setup.py

```diff
@@ -0,0 +1,108 @@
+# SOFTWARE,  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from setuptools import setup
+from os import getenv, path, walk
+
+BASE_PATH = path.abspath(path.dirname(__file__))
+
+URL = "https://github.com/OpenVoiceOS/skill-spelling"
+SKILL_CLAZZ = "SpellingSkill"  # needs to match __init__.py class name
+PYPI_NAME = "ovos-skill-spelling"  # pip install PYPI_NAME
+
+SKILL_AUTHOR, SKILL_NAME = URL.split(".com/")[-1].split("/")
+SKILL_PKG = SKILL_NAME.lower().replace('-', '_')
+PLUGIN_ENTRY_POINT = f'{SKILL_NAME.lower()}.{SKILL_AUTHOR.lower()}={SKILL_PKG}:{SKILL_CLAZZ}'
+
+def get_version():
+    """Find the version of the package"""
+    version = None
+    version_file = path.join(BASE_PATH, "version.py")
+    major, minor, build, alpha = (None, None, None, None)
+    with open(version_file) as f:
+        for line in f:
+            if "VERSION_MAJOR" in line:
+                major = line.split("=")[1].strip()
+            elif "VERSION_MINOR" in line:
+                minor = line.split("=")[1].strip()
+            elif "VERSION_BUILD" in line:
+                build = line.split("=")[1].strip()
+            elif "VERSION_ALPHA" in line:
+                alpha = line.split("=")[1].strip()
+
+            if (major and minor and build and alpha) or "# END_VERSION_BLOCK" in line:
+                break
+    version = f"{major}.{minor}.{build}"
+    if alpha and int(alpha) > 0:
+        version += f"a{alpha}"
+    return version
+
+
+def get_requirements(requirements_filename: str):
+    requirements_file = path.join(BASE_PATH, requirements_filename)
+    with open(requirements_file, "r", encoding="utf-8") as r:
+        requirements = r.readlines()
+    requirements = [r.strip() for r in requirements if r.strip() and not r.strip().startswith("#")]
+
+    for i in range(0, len(requirements)):
+        r = requirements[i]
+        if "@" in r:
+            parts = [p.lower() if p.strip().startswith("git+http") else p for p in r.split("@")]
+            r = "@".join(parts)
+        if getenv("GITHUB_TOKEN"):
+            if "github.com" in r:
+                requirements[i] = r.replace("github.com", f"{getenv('GITHUB_TOKEN')}@github.com")
+    return requirements
+
+
+def find_resource_files():
+    resource_base_dirs = ("locale", "vocab", "dialog", "regex", "sounds", "ui")
+    base_dir = BASE_PATH
+    package_data = ["skill.json"]
+    for res in resource_base_dirs:
+        if path.isdir(path.join(base_dir, res)):
+            for directory, _, files in walk(path.join(base_dir, res)):
+                if files:
+                    package_data.append(path.join(directory.replace(base_dir, "").lstrip("/"), "*"))
+    #    print(package_data)
+    return package_data
+
+
+with open(path.join(BASE_PATH, "README.md"), "r") as f:
+    long_description = f.read()
+
+setup(
+    name=f"{SKILL_NAME}",
+    version=get_version(),
+    url=f"https://github.com/OpenVoiceOS/{SKILL_NAME}",
+    license="Apache2",
```

Comment:
[Apache-2.0](https://spdx.org/licenses/Apache-2.0.html) is the correct SPDX short identifier